pyiron_base/database/generic.py

```diff
@@ -372,6 +372,33 @@ def get_items_sql(self, where_condition=None, sql_statement=None):
             output_list += [dict(zip(col.keys(), tmp_values))]
         return output_list
 
+    def _truncate_chem_formula(self, chem_formula):
+        """
+        this functions truncate the chemical formula to the last element,
+        up to the limit of the database.
+        arg:
+        chem_form(str): the chemical formula
+        """
+        acceptable_len=30
+        if len(chem_formula) >= acceptable_len:
+            for i in range(acceptable_len):
+                if not chem_formula[acceptable_len-i].isdigit():
+                    return chem_formula[:acceptable_len-i+1]
```

Comment:
This doesn't quite work, e.g. 

```python
_truncate_chem_formula("Fe1000B3000X333333G33333444444000000") == 'Fe1000B3000X333333G'
```
but not 
```python
'Fe1000B3000X333333'
```

A hack-ish version that works, but needs `dropwhile` from `itertools`
```suggestion
            if chem_formula[acceptable_len].isalpha() and chem_formula[acceptable_len-1].isdigit():
                return chem_formula[:acceptable_len] # acceptable length happens to be just between two elements
            else:
                return "".join(dropwhile(str.isalpha, dropwhile(str.isdigit, chem_formula[acceptable_len::-1])))[::-1]
```

(check for typos :S)

Reply:
Now I got Liam's suggestion in the meeting, and I changed the formula to `OVERFLOW_ERROR`.